dask/dataframe/io/parquet/core.py

```diff
@@ -0,0 +1,452 @@
+    partition_on : list, optional
+        Construct directory-based partitioning by splitting on these fields'
+        values. Each dask partition will result in one or more datafiles,
+        there will be no global groupby.
+    storage_options : dict, optional
+        Key/value pairs to be passed on to the file-system backend, if any.
+    compute : bool, optional
+        If True (default) then the result is computed immediately. If False
+        then a ``dask.delayed`` object is returned for future computation.
+    **kwargs
+        Extra options to be passed on to the specific backend.
+
+    Examples
+    --------
+    >>> df = dd.read_csv(...)  # doctest: +SKIP
+    >>> to_parquet('/path/to/output/', df, compression='snappy')  # doctest: +SKIP
+
+    See Also
+    --------
+    read_parquet: Read parquet data to dask.dataframe
+    """
+    partition_on = partition_on or []
+    if isinstance(partition_on, string_types):
+        partition_on = [partition_on]
+
+    if set(partition_on) - set(df.columns):
+        raise ValueError("Partitioning on non-existent column")
+
+    if compression != "default":
+        kwargs["compression"] = compression
+    elif "snappy" in compress:
+        kwargs["compression"] = "snappy"
+
+    if isinstance(engine, str):
+        engine = get_engine(engine)
+
+    fs, fs_token, _ = get_fs_token_paths(
+        path, mode="wb", storage_options=storage_options
+    )
+    # Trim any protocol information from the path before forwarding
+    path = infer_storage_options(path)["path"]
+
+    if write_index:
+        df = df.reset_index()
+
+    out = engine.write(
+        df, fs, fs_token, path, append=append, partition_on=partition_on, **kwargs
+    )
+
+    if compute:
+        out.compute()
+        return None
+    return out
+
+
+_ENGINES = {}
+
+
+def get_engine(engine):
+    """Get the parquet engine backend implementation.
+
+    Parameters
+    ----------
+    engine : {'auto', 'fastparquet', 'pyarrow'}, default 'auto'
+        Parquet reader library to use. Default is first installed in this list.
+
+    Returns
+    -------
+    A dict containing a ``'read'`` and ``'write'`` function.
+    """
+    if engine in _ENGINES:
+        return _ENGINES[engine]
+
+    if engine == "auto":
+        for eng in ["fastparquet", "pyarrow"]:
+            try:
+                return get_engine(eng)
+            except RuntimeError:
+                pass
+        else:
+            raise RuntimeError("Please install either fastparquet or pyarrow")
+
+    elif engine == "fastparquet":
+        import_required("fastparquet", "`fastparquet` not installed")
+        from .fastparquet import FastParquetEngine
+
+        _ENGINES["fastparquet"] = eng = FastParquetEngine
+        return eng
+
+    elif engine == "pyarrow" or engine == "arrow":
+        pa = import_required("pyarrow", "`pyarrow` not installed")
+        from .arrow import ArrowEngine
+
+        if LooseVersion(pa.__version__) < "0.8.0":
```

Comment:
Strongly suggest requiring a much more recent version, 0.9.0 is over a year old.

Reply:
Happy to change this.  We'll probably have to figure out what makes sense.  I suggest that we check this out after the test suite is in better shape, and see how far back we can take things comfortably and then pin there.